src/importlinter/contracts/layers.py

```diff
@@ -29,18 +33,38 @@ class Layer:
 
 class LayerField(fields.Field):
     def parse(self, raw_data: str | list) -> Layer:
-        module_tails = set()
         raw_string = fields.StringField().parse(raw_data)
-        raw_items = [item.strip() for item in raw_string.split("|")]
-        for raw_item in raw_items:
-            if raw_item.startswith("(") and raw_item.endswith(")"):
-                module_tail = raw_item[1:-1]
+        raw_module_tails, independent = self._split_raw_module_tails(raw_string)
```

Comment:
It feels like it would be simpler if we broke this into two functions calls, one that tells us whether the layer is independent, and one that splits the elements. The splitter could be agnostic of what the delimiter is and not care if there are two different kinds.

Do you agree?

Reply:
I've split this into two separate method calls now